app/src/main/java/org/simple/clinic/bp/entry/ConfirmRemoveBloodPressureDialog.kt

```diff
@@ -107,4 +115,12 @@ class ConfirmRemoveBloodPressureDialog : AppCompatDialogFragment() {
     super.onDestroyView()
     screenDestroys.onNext(ScreenDestroyed())
   }
+
+  override fun onAttach(context: Context?) {
+    super.onAttach(context)
+    removeBloodPressureListener = context as? RemoveBloodPressureListener
```

Comment:
I am not comfortable doing this because this ties the dialog to be used in an `Activity` instead of a `Screen`. If we ever need to open this from a screen, then the activity the screen lives in will have to implement this even though the dialog itself is opened in the screen and then it will have to pass the event to the screen is some fashion, which will be ugly.

We do have a `ScreenRouter` which exposes a bus for passing events between screens in this fashion. The dialog can push an event to the bus, and regardless of whether it is being invoked in an activity or a screen, the bus can be subscribed to for this event. 

What do you think about using this method instead? We can do this in another PR.

Reply:
There are few things to note here.

1. This is a very specific dialog, used in a very specific feature. Which is a confirm blood pressure dialog, used in a BP entry screen.
2. The underlying UI itself lies inside an `Activity`. Since all our bottom sheets are designed as Activities, we won't be moving away from these unless we are moving away from Flow.
3. The `ScreenRouter` roughly compensates for a coordinator, which we will bring in at some point in time. However, when we start facilitating communication between components that has a hierarchy, our strategy would be to facilitate downstream events via a stream and upstream communication via a callback. Hence, using the `ScreenResultBus` may not be the best solution at the moment. Also, we are losing type-safety by passing across objects of type `Any`.

Reply:
Got it, let's revisit this later.